stdlib/universe/holt_winters/holt_winters.go

```diff
@@ -153,33 +153,24 @@ func (r *HoltWinters) Do(vs *array.Float) *array.Float {
 						panic(err)
 					}
 
-					f := mutable.NewFloat64Array(r.alloc)
-					defer f.Release()
-					f.AppendValues(result.X)
-					sse, newParams := result.F, f
-
-					if sse < minSSE || bestParams == nil {
-						if bestParams != nil {
-							// Previous bestParams are not the best anymore. We can release them.
-							bestParams.Release()
-						}
-						minSSE = sse
-						bestParams = newParams
-					}
-					if bestParams != newParams {
-						// NewParams are not the best. They are useless. Release them.
-						newParams.Release()
+					if result.F < minSSE || bestParams == nil {
+						minSSE = result.F
+						bestParams = result.X
 					}
 				}
 			}
 		}
 	}
 
 	// Final forecast
+	bestParamsF := mutable.NewFloat64Array(r.alloc)
+	bestParamsF.Resize(size)
+	defer bestParamsF.Release()
+	for i := 0; i < len(bestParams); i++ {
+		bestParamsF.Set(i, bestParams[i])
+	}
 	fcast := func() *mutable.Float64Array {
-		fcast := r.forecast(bestParams, false)
-		// Now that bestParams have been used to generate the final forecast, they can be released.
-		defer bestParams.Release()
+		fcast := r.forecast(bestParamsF, false)
 		return fcast
 	}()
```

Comment:
Do we still need this anonymous function? Seems like it was there to ensure the `defer` ran eagerly. Do we still need it?

Reply:
make sense, removed